### PR TITLE
Add Module to model

### PR DESCRIPTION
**Pull request by geraldneo567**
Added Module class and adapted the Model to function with this Module class.
Also added a ModuleBuilder so in future we are able to easily write tests (similar to PersonBuilder).

One more thing to take note: When you face weird errors with working with `Module`, make sure it is imported correctly i.e `import ...module.Module;` because Java has another default library called Module, the IDE always gets confused between those 2.

Things I need you guys to help with:
1) Help me look through for any logical errors or anything I may have overlooked

2) Help me double check if I forgot to change some of the javadoc comments from `Person` to `Module` - I already looked through once and seems fine.

3) In `Module.java` file, is using `List<Person>` for the list of students better, or `List<Student>`?

### CODE REVIEW

File: src/main/java/seedu/address/model/module/Module.java [hunk 1]
```diff
@@ -1,2 +1,127 @@
...
+public class Module {
+
+    // Identity fields
+    private final ModuleName moduleName;
+    private final ModuleCode moduleCode;
+
+
+    // Data fields
+    private final Professor professor;
+    private final List<Student> students;
```
Comment (geraldneo567) [inline src/main/java/seedu/address/model/module/Module.java:26]: Is List<Person> better here?

Comment (benmurphyy) [reply src/main/java/seedu/address/model/module/Module.java:26]: List<Student> better
also make a List<Professor> 
as there can be more than 1 prof per mod

Comment (benmurphyy) [reply src/main/java/seedu/address/model/module/Module.java]: thought about the observable list problem earlier, indeed we need the students and profs of a module to be in a observable list as well, else wont be able to track change in student/profs of a module from gui, so just use ObservableList<Student> and ObservableList<Professor>

**Review by benmurphyy — CHANGES_REQUESTED** [review]
Just change the prof and student list to observable and its good

**Comment by geraldneo567** [comment]
> Just change the prof and student list to observable and its good

ok done

**Review by benmurphyy — APPROVED** [review]

**Pull request merged**